packages/powersync-vue/src/composables/usePowerSyncQuery.ts

```diff
@@ -0,0 +1,56 @@
+import { MaybeRef, Ref, ref, toValue, watch } from 'vue';
+import { usePowerSync } from './powerSync';
+
+export type QueryOptions = { watchParameters: boolean; immediate: boolean };
+export type QueryResult<T> = { data: Ref<T[]>; loading: Ref<boolean>; error: Ref<Error>; refresh: () => Promise<void> };
+
+/**
+ * A composable to access a single static query.
+ * SQL Statement and query Parameters are watched by default.
+ * For a result that updates as the source data changes, use {@link usePowerSyncWatchedQuery} instead.
+ */
+export const usePowerSyncQuery = <T = any>(
+  sqlStatement: MaybeRef<string>,
+  parameters: MaybeRef<any[]> = [],
+  queryOptions: QueryOptions = { watchParameters: true, immediate: true }
+): QueryResult<T> => {
+  const data = ref([]);
+  const loading = ref<boolean>(false);
+  const error = ref<Error>(undefined);
+
+  const powerSync = usePowerSync();
+
+  const fetchData = async () => {
+    if (!powerSync) {
+      error.value = new Error('PowerSync not configured.');
+      return;
+    }
+
+    try {
+      error.value = undefined;
+      loading.value = true;
+
+      await powerSync.value.readLock(async (tx) => {
+        const result = await tx.getAll(toValue(sqlStatement), toValue(parameters));
+        data.value = result;
+      });
```

Comment:
```suggestion
      data.value = await powerSync.value.getAll(toValue(sqlStatement), toValue(parameters));
```

The `getAll` should use a readLock by default.